libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -255,3 +255,10 @@ bool SyntaxChecker::visit(VariableDeclaration const& _declaration)
 	}
 	return true;
 }
+
+bool SyntaxChecker::visit(StructDefinition const& _struct)
+{
+	if (_struct.members().count() == 0)
+		m_errorReporter.syntaxError(_struct.location(), "Defining empty structs is disallowed.");
```

Comment:
I chose the `SyntaxChecker` because the parser is worse at useful error reporting.